site/src/components/DurationField/DurationField.tsx

```diff
@@ -0,0 +1,115 @@
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TextField from "@mui/material/TextField";
+import { useState, type FC } from "react";
+
+type TimeUnit = "days" | "hours";
+
+// Value should be in milliseconds or undefined. Undefined means no value.
+type DurationValue = number | undefined;
+
+type DurationFieldProps = {
+  label: string;
+  value: DurationValue;
+  onChange: (value: DurationValue) => void;
+};
+
+export const DurationField: FC<DurationFieldProps> = (props) => {
+  const { label, value, onChange } = props;
+  const [timeUnit, setTimeUnit] = useState<TimeUnit>(() => {
+    if (!value) {
+      return "hours";
+    }
+
+    return Number.isInteger(durationToDays(value)) ? "days" : "hours";
+  });
+
+  return (
+    <div
+      css={{
+        display: "flex",
+        gap: 8,
+      }}
+    >
+      <TextField
+        css={{ maxWidth: 160 }}
+        label={label}
+        value={
+          !value
+            ? ""
+            : timeUnit === "hours"
+              ? durationToHours(value)
+              : durationToDays(value)
+        }
+        onChange={(e) => {
+          if (e.target.value === "") {
+            onChange(undefined);
+          }
+
+          const value = parseInt(e.target.value);
+
+          if (Number.isNaN(value)) {
+            return;
+          }
+
+          onChange(
+            timeUnit === "hours"
+              ? hoursToDuration(value)
+              : daysToDuration(value),
+          );
+        }}
+        inputProps={{
+          step: 1,
+          type: "number",
+        }}
+      />
+      <Select
+        css={{ width: 120, "& .MuiSelect-icon": { padding: 2 } }}
+        value={timeUnit}
+        onChange={(e) => {
+          setTimeUnit(e.target.value as TimeUnit);
+        }}
+        inputProps={{ "aria-label": "Time unit" }}
+        IconComponent={KeyboardArrowDown}
+      >
+        <MenuItem
+          value="hours"
+          disabled={Boolean(value && !canConvertDurationToHours(value))}
```

Comment:
There's something about these `disabled` expressions that's throwing me for a loop. They're not working how I'd expect them to, but they might still be working as intended

Let's say `value` is either `0` or `undefined` (it doesn't matter – they'll both trigger the behavior), and we're ready to evaluate the `disabled` prop:
1. We need to evaluate everything inside `Boolean()` first, so we start with `value && !canConvertDurationToHours(value)`
2. `value` is falsey because it's `0` or `undefined`, so the entire `&&` expression short-circuits, and the entire expression becomes `0` or `undefined`
3. We then evaluate `Boolean(0)` or `Boolean(undefined)`, and regardless of which one we have, the final expression becomes `false`
4. The new value of the `disabled` prop is false, so both selects are fully functional

Is that intended? My gut feeling would be "no", but I could see the argument that if a value is `0`, it can easily be converted into any other unit

Reply:
Though honestly, just because this component has so many numeric edge cases to deal with, I feel like we'd probably want a Jest component test, too

Reply:
Off the top of my head, I know that we intend for the milliseconds to always be integers, but if a mantissa ever sneaks in (like `3.00000000001`), then I'm not sure if that would break anything

Would we want to use epsilon-based conversions to make sure that we can't actually update state via fractions?

Reply:
Not sure if I got the first point, sorry but I will try to answer with what I was able to understand. When I read the expression for the disabled prop I'm reading as "It is disabled if there is a duration and this duration can't be converted to hours" which for me sounds correct. 

For this component, we can be sure and assume it is always going to receive milliseconds as an integer.

Reply:
For the first point, maybe it's better for me to say that it wasn't clear to me at first what value `disabled` would be. With the current code, I had to stop and think about how logical short-circuiting would play into things

Reply:
For the second point, I get that the value will always be in milliseconds, but even if we intend for them to always be treated as integers, the fact that JS/TS only support floats means there's a risk that more floating-point issues could creep in

Like, anytime we have more involved math calculations, there's a chance that a value we think is supposed to be `3` ends up being `3.0000000001` instead. Still, I think you're right – I don't expect the UI to have to do that many involved math calculations, and that realistically, we'd be directly passing an "integer-like" value that we receive from the server